src/devtools.ts

```diff
@@ -58,8 +62,8 @@ export function preactDevtoolsPlugin({
 		transform(code, url) {
 			const { id } = parseId(url);
 
-			if (entry === id && (!config.isProduction || injectInProd)) {
-				const source = injectInProd ? "preact/devtools" : "preact/debug";
+			if (entry === id && devToolsEnabled) {
+				const source = devtoolsInProd ? "preact/devtools" : "preact/debug";
```

Comment:
I don't quite understand why the source changes based upon whether `devtoolsInProd` is enabled or not. Is this correct? This seems wrong to me, but it's perfectly possible I'm missing something.

Reply:
During production you likely don't want the weight of `preact/debug` with all the warnings, just the devtools connection bridge. That's why the switch was originally introduced. The `preact/devtools` bridge is <200b .

Reply:
Oh shoot, I had it in my mind for some reason that devtools pulled in debug, not the other way around. Didn't even double check when confused.

My bad, cheers!

Reply:
Still, I think this is a bit of a problem as it'd mean enabling it would cause problems with dev, no?

Will correct tomorrow

Reply:
Should be correct now, will always provide `'preact/debug'` in dev if `devToolsEnabled` is set to `true`/default, while in prod, you'll get `'preact/devtools'` only if you've enabled `devtoolsInProd`.